### DISCUSSION TITLE
Rust Desk Change ID

### DISCUSSION THREAD

**Discussion by aabdullaa** [rustdesk/rustdesk#6381] [General]
Dear team,

I am trying to change my ID in the Rust Desk 1.2.3, however it says :"Not yet supported by server"
![Change ID](https://github.com/rustdesk/rustdesk/assets/150546821/188dd70f-ef7c-4e82-a4d4-c12235a32938)

Any solution for this issue?

Thanks

**Comment by akiross** [comment]
I am also looking for a solution to this. I would like to re-generate a random (?) numeric ID, I don't need to define my own, so a random one would be fine.

I found some discussions/issues about this, that suggest to stop the service and all running instances, modify some toml file with the key, but this did not change the situation and the ID is still the same on my windows host. I'd be very happy to have a button to generate a new valid ID automatically.

**Reply by aabdullaa** [reply]
Hello [akiross],

I finally found a solution for this issue.

The issue occurs because Rustdesk do not delete ALL files even after the uninstall.

The solution was by using the free Advanced Uninstaller from: https://www.advanceduninstaller.com/ .

Then uninstall the Rustdesk,using this program, and choose to delete ALL files.

After doing so, re-install the Rustdesk and you will get a new code, and it will work smoothly :)

Please try it, and it will solve your issue.

Best regards,

**Reply by akiross** [reply]
I found out that rustdesk saves config files in *two* places, so I had to edit both. Uninstalled, removed them and reinstalled, worked fine.

**Reply by akiross** [reply]
Just for info, on Windows the directories are in `Users\user\AppData\Local\` and `Users\user\AppData\Roaming`.

**Comment by abdullah-erturk** [comment]
> Sevgili ekip,
> 
> Rust Desk 1.2.3'te kimliğimi değiştirmeye çalışıyorum, ancak "Henüz sunucu tarafından desteklenmiyor" yazıyor. ![Kimliği değiştir](https://private-user-images.githubusercontent.com/150546821/282258704-188dd70f-ef7c-4e82-a4d4-c12235a32938.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4OTQwOTAsIm5iZiI6MTcwOTg5Mzc5MCwicGF0aCI6Ii8xNTA1NDY4MjEvMjgyMjU4NzA0LTE4OGRkNzBmLWVmN2MtNGU4Mi1hNGQ0LWMxMjIzNWEzMjkzOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwOFQxMDI5NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NjAyZjVmZTIwN2VhOWMzZjIxYTA5MTQ4Y2I5ZWFkMDA4ODM5MzU3YWY4Y2IwYTU1ZDNjODM5YWQ4ZTE0ZjQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.IbFeaXD6c-Hsht3xJMg621KDCk-FP-P3pHpVsJnT6to)
> 
> Bu sorun için herhangi bir çözüm var mı?
> 
> Teşekkürler

I prepared a script file for RustDesk to change the ID. The process is very easy and completely automatic, I recommend you try it.

https://github.com/abdullah-erturk/RustDesk-ID-Changer

**Reply by xlionjuan** [reply]
It doesn't work on Windows sandbox while I'm testing, or I should test it on virtual machine?

**Reply by KermitDeF** [reply]
Worked for me! Thank you very much. Had enough of TV claiming commercial use on me and looked up alternatives just today and found RD and saw the option in the GUI and wondered why it didn't work, then found this! Having a unique name is much better than numbers! You're my hero today.

**Reply by robpomeroy** [reply]
> I prepared a script file for RustDesk to change the ID. The process is very easy and completely automatic, I recommend you try it.
> 
> https://github.com/abdullah-erturk/RustDesk-ID-Changer

I've fixed a bug in the script. The fixed version works for me. I've opened a PR to [abdullah-erturk](https://github.com/abdullah-erturk)'s repo. Until that's merged, you can find the fixed version [here](https://github.com/robpomeroy/RustDesk-ID-Server-Changer/blob/bug-program-path/RustDesk_ID_Server_Changer.bat). (And you can [check the diff](https://github.com/abdullah-erturk/RustDesk-ID-Server-Changer/pull/17/changes) to be certain I'm not pulling a fast one. 😉)

He deserves all the credit. Works well.

**Comment by xgusto** [comment]
I have NTB dualboot Linux/Win10.
I wanted to set the same ID on Win10 as I have on Linux, but if the ID already exists, RustDesk immediately generates a new ID.
If I disconnect from the internet and use RustDesk-ID-Changer, it works, but if I connect to the internet, RustDesk immediately generates a new ID.

**Reply by xlionjuan** [reply]
The auto generated ID is based on the SID (Windows) or machine ID (Linux), so it absolutely can't be the same.

**Comment by rijotech** [comment]
I tried to change the ID on linux but I can't find a way.

**Reply by imar4ik** [reply]
.

**Comment by imar4ik** [comment]
stop service rustdesk
change **user**
```
sudo sed -i "s/^enc_id = '.*'/enc_id = '00'/" /home/**user**/.config/rustdesk/RustDesk.toml
sudo rm /etc/machine-id
sudo systemd-machine-id-setup
```
run service
